### PR TITLE
comm_split_type HW_GUIDED fix MPI_UNDEFINED handling

### PR DESCRIPTION
**Pull request by jjhursey**

### CODE REVIEW

**Comment by jjhursey** [comment]
@dalcinl can you take this PR for a spin and let me know if it addresses the issue you mentioned [here]? Thanks!

**Comment by dalcinl** [comment]
@jjhursey Here you have the [logs](https://github.com/mpi4py/mpi4py-testing/runs/7966782746?check_suite_focus=true), I believe everything is OK regarding this PR. 

However, I got PRTE error when trying to run with 5 processes (IIRC, GitHub Action workers have just 2 virtual cores), with oversubscription turned on via `$HOME/.prte/mca-params.conf` [[link]](https://github.com/mpi4py/mpi4py-testing/runs/7966782746?check_suite_focus=true#step:9:7).
```
Run mpiexec -n 5 python mpi4py/test/runtests.py -v -f
[fv-az163-551:165345] PRTE ERROR: Unable to map job in file rmaps_rr.c at line 184
Error: Process completed with exit code 59.
```
This new issue is most likely unrelated to this PR, but anyway you may be the right person to notify. This is probably a regression after the recent PRTE updates.

**Comment by dalcinl** [comment]
@jjhursey Forgot to mention: PRs related to the new comm split types (this one and the previous ones) should be also merged into branch v5.0.x, right? Maybe you should label them appropriately? Or are they already merged?

**Comment by jjhursey** [comment]
For the runtime issue, I added a note to Issue #10698 on which the team is currently looking at these type of issues.

As far as `v5.0`, I'll ask to see if the RMs want to let it in. If so then I'll PR over the full set of fixes and let you know.

**Comment by jjhursey** [comment]
FYI: I created a PR for v5.0.x that cherry-picks all of the changes so far. Once this PR is merged then I'll work with the v5 RMs to get it into the v5.0 release cycle.
 * https://github.com/open-mpi/ompi/pull/10703

**Comment by rhc54** [comment]
> However, I got PRTE error when trying to run with 5 processes (IIRC, GitHub Action workers have just 2 virtual cores), with oversubscription turned on

Should be fixed in https://github.com/openpmix/prrte/pull/1463

**Comment by dalcinl** [comment]
Gentle request to move this PR forward.

**Comment by jjhursey** [comment]
@bosilca this is ready to review when you have a chance. Thanks

File: ompi/communicator/comm.c [hunk 1]
```diff
@@ -941,26 +941,34 @@ int ompi_comm_split_type (ompi_communicator_t *comm, int split_type, int key,
...
             if (OPAL_UNLIKELY(OMPI_SUCCESS != rc)) {
                 return rc;
             }
         }
 
-        if (OPAL_UNLIKELY(!ok)) {
+        if (OPAL_UNLIKELY(!ok[0] || !ok[1])) {
+            if (0 == ompi_comm_rank(comm)) {
+                if (!ok[1]) {
+                    opal_output(0, "Error: Mismatched info values for MPI_COMM_TYPE_HW_GUIDED");
```
Comment (bosilca) [inline ompi/communicator/comm.c:967]: increase the user-friendliness by using opal_show_help here.

Comment (jjhursey) [reply ompi/communicator/comm.c]: I made the change. Let me know what you think.

**Review by bosilca — APPROVED** [review]

**Pull request merged**